c7n/filters/vpc.py

```diff
@@ -29,6 +30,24 @@ class SecurityGroupFilter(RelatedResourceFilter):
     AnnotationKey = "matched-security-groups"
 
 
+class VpcFilter(RelatedResourceFilter):
+    """Filter VPCs by associated security groups."""
+    schema = type_schema(
+        'security-group', rinherit=ValueFilter.schema,
+        **{'match-resource':{'type': 'boolean'},
+           'operator': {'enum': ['and', 'or']}})
+
+    RelatedResource = "c7n.resources.vpc.SecurityGroup"
+    AnnotationKey = "matched-vpcs"
+
+    def get_related_ids(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
+        vpc_ids = [vpc['VpcId'] for vpc in resources]
+        vpc_filter = {'Name': 'vpc-id', 'Values': vpc_ids}
+        response = client.describe_security_groups(Filters=[vpc_filter])
```

Comment:
we're better off just pulling the security groups from cache via the resource manager (can pull from multiple sources as well that way).

ie 

vpc_groups = [g for g in self.manager.get_resource_manager('security-group').resources() if g.get('VpcId', '') in vpc_ids]